src/device/controller/inputconvert/inputconvertgame.cpp

```diff
@@ -493,11 +494,31 @@ void InputConvertGame::processKeyDrag(const QPointF &startPos, QPointF endPos, c
         m_dragDelayData.currentPos = startPos;
         m_dragDelayData.queuePos.clear();
         m_dragDelayData.queueTimer.clear();
+
+        // Clamp dragSpeed to 0-1 range
+        const float speed = qBound(0.0f, static_cast<float>(dragSpeed), 1.0f);
+        
+        // Calculate delays based on dragSpeed
+        // dragSpeed = 1 -> minDelay = 1, maxDelay = 2 (fastest)
+        // dragSpeed = 0 -> minDelay = 30, maxDelay = 40 (slowest)
+        const quint32 minDelay = static_cast<quint32>(1 + (1.0f - speed) * 29);  // 1 to 30
+        const quint32 maxDelay = minDelay + static_cast<quint32>((1.0f - speed) * 9) + 1;  // // min + (0 to 9) + 1
+
         getDelayQueue(startPos, endPos,
-                      0.01f, 0.002f, 0, 2,
+                      0.01f, 0.0005f,
+                      minDelay,
+                      maxDelay,
                       m_dragDelayData.queuePos,
                       m_dragDelayData.queueTimer);
-        m_dragDelayData.timer->start();
+
```

Comment:
just add startDelay param is better?
 m_dragDelayData.timer->start(startDelay);

Reply:
I am not a C/C++ developer, but I tried to implement the feature as best as I could.
Thank you for the awesome support <3

PS. I am working on a visual mapper for the keybindings, I'll release it soon enough for the public
![image](https://github.com/user-attachments/assets/fe997417-6f8d-4fb9-9d52-86e36f2a42e5)

Reply:
I will help you review the code. Thank you for your contribution
Looking forward to your visual mapper. Can it be pr to QtScrcpy？

Reply:
Thank you <3
It is a website actually, since I am a C# developer and not C/C++ I couldn't add such a feature to be natively on the QtScrcpy itself.

I have just published the repo: https://github.com/w4po/ScrcpyKeyMapper
live: https://w4po.github.io/ScrcpyKeyMapper/

Reply:
> just add startDelay param is better? m_dragDelayData.timer->start(startDelay);

fix this?

Reply:
Oh I thought you did it already :(
My bad, I am still learning the Github stuff.
Done now?